crates/primitives/src/result.rs

```diff
@@ -135,6 +135,11 @@ pub enum EVMError<DBError> {
     Header(InvalidHeader),
     /// Database error.
     Database(DBError),
+    /// Custom error.
+    ///
+    /// Currently implemented for the Optimism handler. This is an interim solution pending
+    /// the development of a more generic error mechanism for integration with custom handlers.
```

Comment:
Would change this comment to reflect the present state and not a possible/unknown future.

```suggestion
    /// Useful for handler registers where custom logic would want to return their own custom error.
```